src/FSharp.Data.GraphQL.Shared/TypeSystem.fs

```diff
@@ -1115,7 +1122,60 @@ and [<CustomEquality; NoComparison>] ScalarDefinition<'Primitive, 'Val> = {
 
 and ScalarDefinition<'Val> = ScalarDefinition<'Val, 'Val>
 
-/// A GraphQL representation of single case of the enum type.
+and FileDef =
+    interface
+        /// Name of the file type.
+        abstract Name : string
+        /// Optional scalar type description.
+        abstract Description : string option
+        /// A function used to retrieve a .NET object from provided GraphQL query or JsonElement variable.
+        abstract Coerce : IInputExecutionContext -> InputParameterValue -> Result<obj, string>
+        inherit TypeDef
+        inherit NamedDef
+        inherit InputDef
+    end
+
+and [<CustomEquality; NoComparison>] FileDefinition = {
```

Comment:
I thougth you use `InputCustom` and you do not need a separate definition for a file